src/frontends/pytorch/src/translate_session.cpp

```diff
@@ -257,6 +294,111 @@ size_t TranslateSession::decode_tensor_name(const Output<Node>& output) {
     return static_cast<size_t>(std::stoll(name));
 }
 
+namespace {
+Output<Node> slice_backprop(std::shared_ptr<TorchDecoder> node, Output<Node> slice_output, Output<Node> value) {
+    auto slice_node = slice_output.get_node_shared_ptr();
+    FRONT_END_OP_CONVERSION_CHECK(ov::as_type_ptr<v8::Slice>(slice_node),
+                                  "Conversion rule for aten::slice doesn't contain Slice node.");
+
+    auto zero = v0::Constant::create(element::i64, Shape{}, {0});
+    auto one = v0::Constant::create(element::i64, Shape{}, {1});
+    auto neg_one_1d = v0::Constant::create(element::i64, Shape{1}, {-1});
+    auto scattering_shape = v0::Constant::create(element::i64, Shape{2}, {-1, 1});
+
+    // Get 1d indices [0..numel)
+    auto to_insert_data = slice_node->input_value(0);
+    auto input_shape = std::make_shared<v3::ShapeOf>(to_insert_data, element::i64);
+    auto numel = std::make_shared<v1::ReduceProd>(input_shape, zero, false);
+    auto full_data_indices_1d = std::make_shared<v4::Range>(zero, numel, one, element::i64);
+
+    // Slice indices by same start, stop, slice, axes as initial Slice
+    auto full_data_indices = std::make_shared<v1::Reshape>(full_data_indices_1d, input_shape, false);
+    Output<Node> data_indices;
+    if (slice_node->get_input_size() == 5) {
+        data_indices = std::make_shared<v8::Slice>(full_data_indices,
+                                                   slice_node->input_value(1),
+                                                   slice_node->input_value(2),
+                                                   slice_node->input_value(3),
+                                                   slice_node->input_value(4));
+    } else if (slice_node->get_input_size() == 4) {
+        data_indices = std::make_shared<v8::Slice>(full_data_indices,
+                                                   slice_node->input_value(1),
+                                                   slice_node->input_value(2),
+                                                   slice_node->input_value(3));
+    } else {
+        FRONT_END_OP_CONVERSION_CHECK(false, "Incorrect number of Slice inputs");
+    }
+
+    // Scatter in flattened tensor with indices and flattened data to be inserted
+    auto to_insert_data_1d = std::make_shared<v1::Reshape>(to_insert_data, neg_one_1d, false);
+    auto data_indices_1d = std::make_shared<v1::Reshape>(data_indices, scattering_shape, false);
+    auto to_be_inserted_data_1d = std::make_shared<v1::Reshape>(value, neg_one_1d, false);
+    auto updated_data_1d =
+        std::make_shared<v3::ScatterNDUpdate>(to_insert_data_1d, data_indices_1d, to_be_inserted_data_1d);
+
+    // Reshape to initial shape
+    return std::make_shared<v1::Reshape>(updated_data_1d, input_shape, false);
+}
+
+Output<Node> select_backprop(std::shared_ptr<TorchDecoder> node, Output<Node> select_output, Output<Node> value) {
```

Comment:
I see that `std::shared_ptr<TorchDecoder> node` is not used at all. Can we remove it?

Reply:
Also, I would recommend to use `const Output<Node>& value` to avoid copies

Reply:
The same comment for other function